code/modules/mob/living/carbon/alien/diona/diona_powers.dm

```diff
@@ -420,3 +420,15 @@
 			T.ChangeTurf(structure_path)
 		else
 			new structure_path(T)
+			
+/mob/living/carbon/alien/diona/proc/remove_hat()
+	set category = "Abilities"
+	set name = "Remove Hat"
+	set desc = " Remove your hat."
+	
+	if(hat != null)
+		src.drop_from_inventory(hat)
+		hat = null
+		visible_message("<span class='warning'>[src] removes their hat!</span>")
+	else
+		to_chat(src, span("warning", "You have no hat!"))
```

Comment:
Better way to do this:

```
if(hat)
	...
else
	...
```
Also, doesn't drop_from_inventory already null out the variable?

Reply:
It does not @NonQueueingMatt. I tested it, without hat = null the hat would not actually be removed.